### PR TITLE
FIX-#6582: avoid `FutureWarning`s in `bfill/backfill/ffill/pad` unless necessary

### LINKED ISSUE #6582: Using `bfill` func leads to getting `FutureWarning: DataFrame.fillna with 'method' is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.`

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

~Blocked on #6582~

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6582 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/base.py [hunk 1]
```diff
@@ -1625,6 +1669,13 @@ def fillna(
         Series, DataFrame or None
             Object with missing values filled or None if ``inplace=True``.
         """
+        if method is not None:
+            warnings.warn(
```
Comment (anmyachev) [inline modin/pandas/base.py:1673]: Missed warning

File: modin/pandas/base.py [hunk 2]
```diff
@@ -1564,11 +1586,33 @@ def ffill(
...
+            )
 
-    pad = ffill
+    def pad(
+        self, *, axis=None, inplace=False, limit=None, downcast=lib.no_default
+    ):  # noqa: PR01, RT01, D200
+        """
+        Synonym for `DataFrame.ffill`.
+        """
+        warnings.warn(
```
Comment (anmyachev) [inline modin/pandas/base.py:1607]: Missed warning

File: modin/pandas/base.py [hunk 3]
```diff
@@ -1098,11 +1098,33 @@ def bfill(
...
+            )
 
-    backfill = bfill
+    def backfill(
+        self, *, axis=None, inplace=False, limit=None, downcast=lib.no_default
+    ):  # noqa: PR01, RT01, D200
+        """
+        Synonym for `DataFrame.bfill`.
+        """
+        warnings.warn(
```
Comment (anmyachev) [inline modin/pandas/base.py:1119]: Missed warning

File: modin/pandas/base.py [hunk 4]
```diff
@@ -1564,11 +1586,33 @@ def ffill(
...
+    ):  # noqa: PR01, RT01, D200
+        """
+        Synonym for `DataFrame.ffill`.
+        """
+        warnings.warn(
+            "DataFrame.pad/Series.pad is deprecated. Use DataFrame.ffill/Series.ffill instead",
+            FutureWarning,
+        )
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", category=FutureWarning)
```
Comment (anmyachev) [inline modin/pandas/base.py:1612]: Filter warnings to align with pandas.

File: modin/pandas/base.py [hunk 5]
```diff
@@ -1564,11 +1586,33 @@ def ffill(
         Synonym for `DataFrame.fillna` with ``method='ffill'``.
         """
         downcast = self._deprecate_downcast(downcast, "ffill")
-        return self.fillna(
-            method="ffill", axis=axis, limit=limit, downcast=downcast, inplace=inplace
-        )
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
```
Comment (anmyachev) [inline modin/pandas/base.py:1590]: Filter warnings to align with pandas.

File: modin/pandas/base.py [hunk 6]
```diff
@@ -1098,11 +1098,33 @@ def bfill(
...
+    ):  # noqa: PR01, RT01, D200
+        """
+        Synonym for `DataFrame.bfill`.
+        """
+        warnings.warn(
+            "DataFrame.backfill/Series.backfill is deprecated. Use DataFrame.bfill/Series.bfill instead",
+            FutureWarning,
+        )
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", category=FutureWarning)
```
Comment (anmyachev) [inline modin/pandas/base.py:1124]: Filter warnings to align with pandas.

**Review by dchigarev — APPROVED** [review]

**Comment by anmyachev** [comment]
@mvashishtha do you agree to merge it?

**Comment by dchigarev** [comment]
Ideally, I would like to see the warnings removal by re-routing the `bfill/...` calls from the public version of `.fillna()` method (that raises warnings on the 'method' parameter) to the private `._fillna()` (that doesn't do so)
```python
def bfill(...):
    return self._fillna(method="bfill")

def _fillna(self, value=None, method=None, ...):
    # doesn't raise warnings on the 'method' parameter
    return self._query_compiler.fillna(...)

def fillna(self, value=None, method=None, ...):
    if method is not None:
        warnings.raise("...")
    return self._fillna(...)
``` 
this way we wouldn't silence any warnings and won't need to change anything after the 'method' parameter will be finally removed from the public `.fillna()`

**Comment by anmyachev** [comment]
> Ideally, I would like to see the warnings removal by re-routing the `bfill/...` calls from the public version of `.fillna()` method (that raises warnings on the 'method' parameter) to the private `._fillna()` (that doesn't do so)
> 
> ```python
> def bfill(...):
>     return self._fillna(method="bfill")
> 
> def _fillna(self, value=None, method=None, ...):
>     # doesn't raise warnings on the 'method' parameter
>     return self._query_compiler.fillna(...)
> 
> def fillna(self, value=None, method=None, ...):
>     if method is not None:
>         warnings.raise("...")
>     return self._fillna(...)
> ```
> 
> this way we wouldn't silence any warnings and won't need to change anything after the 'method' parameter will be finally removed from the public `.fillna()`
> 
> UPD: but since the changes are already merged... I think the current approach is also good, to quickly silence the warnings before the release

Makes sense

**Pull request merged**